### ISSUE TITLE
ImportError when running tests.

### ISSUE THREAD

**Issue by Thom1729** [SublimeText/UnitTesting#93]
After [this commit](https://github.com/SublimeText/UnitTesting/commit/c655a6f90e186a6822c1a0c1515af6154a550b8c), the tests fail:

```
test_3141596 (unittest.loader.ModuleImportFailure) ... ERROR

======================================================================
ERROR: test_3141596 (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./python3.3/unittest/case.py", line 384, in _executeTestPart
  File "./python3.3/unittest/loader.py", line 32, in testFailure
ImportError: Failed to import test module: test_3141596
Traceback (most recent call last):
  File "./python3.3/unittest/loader.py", line 261, in _find_tests
  File "./python3.3/unittest/loader.py", line 239, in _get_module_from_name
  File "/home/thom/.config/sublime-text-3/Packages/UnitTesting/tests/test_3141596.py", line 5, in <module>
    from unittesting.utils import UTSetting
ImportError: No module named 'unittesting.utils'


----------------------------------------------------------------------
Ran 1 test in 0.113s

FAILED (errors=1)

UnitTesting: Done.
```

**Comment by randy3k** [comment]
But we don't see failures on travis and circleci, plus, and it works fine on my computer...

I think I really need to study how submodules are imported.

**Comment by Thom1729** [comment]
Have you tried restarting Sublime and running the tests again?

**Comment by randy3k** [comment]
Yap. The tests pass on travis and circleci. May it be an issue for specific platforms?

**Comment by Thom1729** [comment]
Possibly; I'm on Linux. I also have the repo checked out in Packages, if that makes a difference.

**Comment by randy3k** [comment]
So, running `import unittesting.utils` in the Sublime console does not work for you?

**Comment by Thom1729** [comment]
ImportError: No module named 'unittesting.utils'.

**Comment by Thom1729** [comment]
What do you have in `sys.path`?

**Comment by Thom1729** [comment]
Also, immediately after you restart, is `'unittesting.utils' in sys.modules`?

**Comment by randy3k** [comment]
I am on mobile now. But I believe it should be kept being untouched (except for coverage)

**Comment by randy3k** [comment]
Ai, there could be a chance that your platform has mixed up `UnitTesting` and `unittesting`.

**Comment by randy3k** [comment]
I've just check, `sys.path` contains nothing related to `UnitTesting`.

**Comment by Thom1729** [comment]
Try:

```
import importlib
importlib.find_loader('unittesting').load_module('unittesting.utils')
```

**Comment by randy3k** [comment]
I am not sure if it is supposed to work. For instance,
```
importlib.find_loader('os').load_module('os.path')
```
fails to run.

**Comment by Thom1729** [comment]
What's the value of `unittesting.utils.__loader__`?

**Comment by Thom1729** [comment]
(The loader for `os` is a different kind of loader object than for `unittesting`.)

**Comment by randy3k** [comment]
after restarting sublime
```
>>> import unittesting.utils
>>> unittesting.utils.__loader__
<_frozen_importlib.SourceFileLoader object at 0x110277250>
```

We are seeing different loaders since Sublime ships a binary version of Python. It is the same loader (SourceFileLoader) on a regular python.

**Comment by Thom1729** [comment]
How about:

    [ (m, m.find_module('unittesting.utils')) for m in sys.meta_path ]

And for comparison:

    [ (m, m.find_module('UnitTesting')) for m in sys.meta_path ]

If there are any platform-related differences, this should find them. My outputs for those are:

    [(<sublime_plugin.MultizipImporter object at 0x7f8b8ac9bb50>, None), (<class '_frozen_importlib.BuiltinImporter'>, None), (<class '_frozen_importlib.FrozenImporter'>, None), (<class '_frozen_importlib.PathFinder'>, None)]

and:

    [(<sublime_plugin.MultizipImporter object at 0x7f8b8ac9bb50>, <sublime_plugin.ZipLoader object at 0x7f8b8abaec50>), (<class '_frozen_importlib.BuiltinImporter'>, None), (<class '_frozen_importlib.FrozenImporter'>, None), (<class '_frozen_importlib.PathFinder'>, <_frozen_importlib.NamespaceLoader object at 0x7f8b89e38910>)]

If `unittesting.utils` isn't directly patched into `sys.modules`, then it has to be one of the finders in `sys.meta_path` providing the loader. We can see the source code for sublime_plugin.MultizipImporter, and the other three are standard, so hopefully this should get us to the bottom of this issue.

**Comment by randy3k** [comment]
```
>>> [ (m, m.find_module('unittesting.utils')) for m in sys.meta_path ]
[(<sublime_plugin.MultizipImporter object at 0x1053e2590>, None), (<class '_frozen_importlib.BuiltinImporter'>, None), (<class '_frozen_importlib.FrozenImporter'>, None), (<class '_frozen_importlib.PathFinder'>, None), (<requests.packages.urllib3.packages.six._SixMetaPathImporter object at 0x1055d38d0>, None)]
Package Control: Skipping automatic upgrade, last run at 2018-04-11 15:12:16, next run at 2018-04-11 16:12:16 or after
>>> [ (m, m.find_module('unittesting')) for m in sys.meta_path ]
[(<sublime_plugin.MultizipImporter object at 0x1053e2590>, None), (<class '_frozen_importlib.BuiltinImporter'>, None), (<class '_frozen_importlib.FrozenImporter'>, None), (<class '_frozen_importlib.PathFinder'>, None), (<requests.packages.urllib3.packages.six._SixMetaPathImporter object at 0x1055d38d0>, None)]
```

None of the finders find `unittesting` or `unittesting.utils`.

```
>>> [ (m, m.find_module('UnitTesting')) for m in sys.meta_path ]
[(<sublime_plugin.MultizipImporter object at 0x1053e2590>, None), (<class '_frozen_importlib.BuiltinImporter'>, None), (<class '_frozen_importlib.FrozenImporter'>, None), (<class '_frozen_importlib.PathFinder'>, <_frozen_importlib.NamespaceLoader object at 0x10799db50>), (<requests.packages.urllib3.packages.six._SixMetaPathImporter object at 0x1055d38d0>, None)]
```
`UnitTesting` is found by a PathFinder.

**Comment by Thom1729** [comment]
Good news and bad news. It looks like it works when uncompressed, but not when you zip it up as a `.sublime-package`. I nuked my installed version, and it worked; then, I zipped the installed version into `Installed Packages/UnitTesting.sublime-package`, and it doesn't. I can `import unittesting`, but `import unittesting.utils` fails. You get the same behavior if you simply install the stable version from Package Control.
